### ISSUE TITLE
[ARCHETYPE-483] Building a new 2.X archetype with JarMojo writes out a nonsensical warning

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-archetype#762]
**[Tomáš Zilvar](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zizithefirst)** opened **[ARCHETYPE-483](https://issues.apache.org/jira/browse/ARCHETYPE-483?redirect=false)** and commented

When packaging a new style 2.0 custom archetype JAR with JarMojo, the build issues a counter-intuitive warning:

```
[WARNING] Building an Old (1.x) Archetype: consider migrating it to current 2.x Archetype.
```

I suppose that it is because of an inverted if condition in JarMojo.checkArchetypeFile(File file)

```
package org.apache.maven.archetype.mojos;
//...
/**
 * Build a JAR from the current Archetype project.
 *
 * @author rafale
 */
@Mojo( name = "jar", defaultPhase = LifecyclePhase.PACKAGE, requiresProject = true )
public class JarMojo extends AbstractMojo
{
    //...
     private void checkArchetypeFile( File jarFile ) throws MojoExecutionException
     {
         try
         {
             if ( archetypeArtifactManager.isFileSetArchetype( jarFile ) )
             {
                 checkFileSetArchetypeFile( jarFile );
             }
             else if ( !archetypeArtifactManager.isOldArchetype( jarFile ) )
             {
                 getLog().warn( "Building an Old (1.x) Archetype: consider migrating it to current 2.x Archetype." );
             }
             else
             {
                 throw new MojoExecutionException( "The current project does not built an archetype" );
             }
         }
         catch ( UnknownArchetype ua )
         {
             throw new MojoExecutionException( ua.getMessage(), ua );
         }
}    
    //...
}
```

The condition `!archetypeArtifactManager.isOldArchetype( jarFile )` is negated, therefore it issues the warning if the jar **is not** an old archetype, which makes it downright confusing for new users trying to build a new archetype. On the other hand, when the archetype actually **is** old, it does not warn at all.

---

**Affects:** 2.3

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

how was this issue fixed in 2.4, please?
was there a related commit or is it just cleaning of an old issue?

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

ok, found the commit
[ad3e8681708c30ff3ea434100815df756866c8e7](https://git-wip-us.apache.org/repos/asf?p=maven-archetype.git;a=commit;h=ad3e8681708c30ff3ea434100815df756866c8e7)

you should start comment with the Jira issue id (as Maven git conventions http://maven.apache.org/developers/conventions/git.html )
with these, Jenkins will add a comment with the link to the issue

now, please add such a link by hand to recent issues: you won't forget the convention ;)
